### PR TITLE
Removed legacy code and some minor refactoring

### PR DESCRIPTION
**Pull request by seriva**
- Removed legacy code for Epiphany < 0.3.
- Refactored build_saver.py and renamed to build_io.py as we do not only save.
- Updated tests to reflect all changes.
- Also added missing changelogs for #2643, #2666
- Added fixes for #2497 which where outstanding in #2652 and reported in: https://github.com/epiphany-platform/epiphany/issues/2497#issuecomment-938599380

### CODE REVIEW

File: cli/engine/ansible/AnsibleInventoryUpgrade.py [hunk 1]
```diff
@@ -70,38 +68,11 @@ def upgrade(self):
...
-
-            # remove linux and reboot roles if present
-            self.delete_role(new_inventory, 'linux')
-            self.delete_role(new_inventory, 'reboot')
-        else:
-            self.logger.info('Upgrading Ansible inventory Epiphany => 0.3.0')
-
-            # load cluster model from manifest
-            self.manifest_docs = load_manifest_docs(self.backup_build_dir)
-            self.cluster_model = select_single(self.manifest_docs, lambda x: x.kind == 'epiphany-cluster')
```
Comment (atsikham) [inline cli/engine/ansible/AnsibleInventoryUpgrade.py:104]: ❤️

**Review by atsikham — DISMISSED** [review]
LGTM

File: cli/engine/ansible/AnsibleInventoryUpgrade.py (hunk 1 above)
Comment (seriva) [reply cli/engine/ansible/AnsibleInventoryUpgrade.py:104]: Yeah I know, that is for pre 0.3 clusters, first version of Epiphany which where deployed with bash only:D

**Review by atsikham — DISMISSED** [review]

**Review by sbbroot — DISMISSED** [review]

**Review by erzetpe — DISMISSED** [review]

**Review by atsikham — DISMISSED** [review]

**Review by atsikham — APPROVED** [review]

**Review by przemyslavic — APPROVED** [review]

**Comment by przemyslavic** [comment]
✔️ `init` (`--full`)
✔️`apply` + re-`apply`
✔️`backup`
✔️`recovery`
✔️`prepare`
✔️`test`
✔️`upgrade` from v1.1 to dev 
✔️`delete`

**Pull request merged**